web/packages/core/src/ruffle-player.ts

```diff
@@ -1421,10 +1458,25 @@ export class RufflePlayer extends HTMLElement {
         }\n`;
     }
 
+    private hidePreloader(): void {
+        this.preloader.style.display = "none";
+        this.unmuteOverlay.classList.remove("hidden");
+        this.playButton.classList.remove("hidden");
+    }
+
+    private showPreloader(): void {
+        this.preloader.style.display = "flex";
+        this.unmuteOverlay.classList.add("hidden");
+        this.playButton.classList.add("hidden");
+    }
+
     private setMetadata(metadata: MovieMetadata) {
         this._metadata = metadata;
         // TODO: Switch this to ReadyState.Loading when we have streaming support.
         this._readyState = ReadyState.Loaded;
+        if (this.hasPreloader) {
```

Comment:
I think `this.hidePreloader()` can be called unconditionally, as already done in `panic()`.